### DISCUSSION TITLE
UX: improve UI (shopinvader, products...)

### DISCUSSION THREAD

**Discussion by hparfr** [shopinvader/odoo-shopinvader#1295] [Ideas]
It's hard to figure if a product is binded (A), if its binding is up to date (recomputed ? exported ?) (B), when was the last sync (B), is it queued ?
After editing a product, there is no feedback to the user about the next steps. (C)

Suggestions:
* implement last_synchronization date on shopinvader_variant (B)
* add filters or menu entrees in Sales > Products , or in PIM (A)
* indicate on the product that it will be recomputed or exported in the future  ? (C)
...

**Comment by hparfr** [comment]
SHOPINVADER 
UX CLEANUP

- Product binding  (publishing)
- Shopinvader backend « config panel »
	- [ ] move company from Sale to Main (tab) -> @pabloeforgeflow
        - [X] sale tab : Notification rename to « internal notification » and add help message -> @pabloeforgeflow #1304 
 	- [X] Add help message on « Visible delivery order states » -> #1324
	- [X] Remove nolabel to carrier_ids  -> https://github.com/shopinvader/odoo-shopinvader/pull/1324
	- [X] SE : se_backend_id : add a label or put it somewherelese ? #1304
    - [ ] Use sale profile » and sale_profile_ids should be next on of the other
	(move currency), Priclelist_id should be labelled « default price list » when sale profile is in use
	customer default rôle : add as explannation and a label
     - [X] Notification  tab : help message describing what are notifications #1325


- [ ] Elastic index creation → discussed in issue https://github.com/OCA/search-engine/issues/111
- [X] Elastic index settings → implement a chatter + field tracking on body_str ? -> @chafique-delli  https://github.com/OCA/search-engine/pull/126/files
- From Sale > Products add « shopinvader binded » filters @bguillot 
- From Shopinvader > Product, add filter « to be sync » / waiting sync / not online yet ? -> @bguillot 

- From Shopinvader backend : no feed back when «recompute » « export » are hit.
No warning when « clear » is hit.:are you sure ? Please solve this captacha 

Shopinvader > Products  > { Products, Variants } rename to :
Shopinvader > Products > Shopinvader Procucts, Shopinvader Variants ?
Or go to real product instead of shopinvader.product

- [X] Shopinvader Variant : date Modified, date sync are not written -> @HviorForgeFlow  https://github.com/OCA/search-engine/pull/127

Shopinvader binding : from shopinvader menu or from Sale > products (or both?)


POC: implement description (short / long) on sale_channel_* 

Removal of products :
Needs :
- disable the ability to buy the product (like out of stock / not replenished temporary) but still have the page on the website
- hide the product from search but still keep the product page online
- remove the product from the website but keep the description on odoo
- what happen when the product is active = False or unlinked (in odoo)
@earcel19

**Reply by earcel19** [reply]
Need 1 :  disable the ability to buy the product (like out of stock / not replenished temporary) but still have the page on the website
-> The product will be visible but it will not be possible to add it to a cart
Proposal : add a bollean field "Sealable on Shopinvader" and expose it in the index
If true -> ok, the product is sealable 
If false -> Prevent the add to cart

Need 2 : hide the product from search but still keep the product page online
-> usefull to keep the page referenced on the site (SEO...)
Proposal : add a boolean field "visible on Shopinvader" + a date field "Visibility start date" + create a daily cron 
If true -> ok, the product is visible on the shop
If false (without visibility start date) -> the product is not visible on the shop
If false + starting date -> the cron will set the  "Visible on Shopinvader" field to true at the starting date  

Need 3  : remove the product from the website but keep the description on odoo
-> how to delete a product and avoid orphan products ...WIP

Need 4 :  what happen when the product is active = False or unlinked (in odoo)
-> it archives the Shopinvader binding but your product still be visible until the next update of the index.
Unarchiving the product WILL NOT unarchive the Shopinvader binding... WIP
